src/modules/function/declaration.rs

```diff
@@ -98,24 +101,42 @@ impl SyntaxModule<ParserMetadata> for FunctionDeclaration {
                     break
                 }
                 let is_ref = token(meta, "ref").is_ok();
+                let name_token = meta.get_current_token();//saving for possible error
                 let name = variable(meta, variable_name_extensions())?;
                 // Optionally parse the argument type
+                let mut arg_type = Type::Generic;
                 match token(meta, ":") {
                     Ok(_) => {
                         self.arg_refs.push(is_ref);
                         self.arg_names.push(name.clone());
-                        self.arg_types.push(parse_type(meta)?);
+                        arg_type = parse_type(meta)?;
+                        self.arg_types.push(arg_type.clone());
                     },
                     Err(_) => {
                         self.arg_refs.push(is_ref);
                         self.arg_names.push(name.clone());
                         self.arg_types.push(Type::Generic);
                     }
                 }
-                let tok = meta.get_current_token();
-                if token(meta, "=").is_ok() {
-                    return error!(meta, tok, "Default values for function arguments are not yet supported")
+                match token(meta,"="){
```

Comment:
Sorry for nitpicking. I just prefer to keep everything spaced out evenly
```suggestion
                match token(meta,"=") {
```